src/main/java/com/zufar/onlinestore/security/configuration/SpringSecurityConfiguration.java

```diff
@@ -46,6 +54,21 @@ public SecurityFilterChain securityFilterChain(final HttpSecurity httpSecurity,
                 .build();
     }
 
+    @Bean
+    public UserDetailsService userDetailsService(final UserRepository userRepository,
+                                                 final PasswordEncoder passwordEncoder) {
+        return username -> {
+            UserEntity user = userRepository.findUserByUsername(username);
+            if (user == null) {
+                log.warn("Failed to get the user with the username = {}.", username);
+                throw new UsernameNotFoundException(username);
+            }
+            user.setPassword(passwordEncoder.encode(user.getPassword()));
```

Comment:
Is it encoding a password and returning a response with a password, but encoded? Why is the password needed here?